src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,283 @@
+        bool RestartNeeded();
+        bool IsAutoLogonConfigured();
+    }
+
+    public class AutoLogonConfigurationManager : AgentService, IAutoLogonConfigurationManager
+    {
+        private ITerminal _terminal;
+
+        public override void Initialize(IHostContext hostContext)
+        {
+            base.Initialize(hostContext);
+            _terminal = hostContext.GetService<ITerminal>();
+        }
+
+        public void Configure(CommandSettings command)
+        {
+            AssertAdminAccess(false);
+
+            var logonAccount = command.GetAutoLogonUserName();
+
+            string domainName;
+            string userName;
+
+            GetAccountSegments(logonAccount, out domainName, out userName);
+
+            if ((string.IsNullOrEmpty(domainName) || domainName.Equals(".", StringComparison.CurrentCultureIgnoreCase)) && !logonAccount.Contains("@"))
+            {
+                logonAccount = String.Format("{0}\\{1}", Environment.MachineName, userName);
+            }
+
+            Trace.Info("LogonAccount after transforming: {0}, user: {1}, domain: {2}", logonAccount, userName, domainName);
+
+            string logonPassword = string.Empty;
+            var windowsServiceHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            while (true)
+            {
+                logonPassword = command.GetWindowsLogonPassword(logonAccount);
+                if (windowsServiceHelper.IsValidCredential(domainName, userName, logonPassword))
+                {
+                    Trace.Info("Credential validation succeeded");
+                    break;
+                }
+                
+                if (command.Unattended)
+                {
+                    throw new SecurityException(StringUtil.Loc("InvalidWindowsCredential"));
+                }
+                    
+                Trace.Info("Invalid credential entered");
+                _terminal.WriteLine(StringUtil.Loc("InvalidWindowsCredential"));
+            }
+
+            bool isCurrentUserSameAsAutoLogonUser = windowsServiceHelper.HasActiveSession(domainName, userName);                
+            var securityIdForTheUser = windowsServiceHelper.GetSecurityId(domainName, userName);
+            var regManager = HostContext.GetService<IWindowsRegistryManager>();
+            
+            AutoLogonRegistryManager regHelper = isCurrentUserSameAsAutoLogonUser 
+                                                ? new AutoLogonRegistryManager(regManager)
+                                                : new AutoLogonRegistryManager(regManager, securityIdForTheUser);
+            
+            if(!isCurrentUserSameAsAutoLogonUser && !regHelper.ValidateIfRegistryExistsForTheUser(securityIdForTheUser))
+            {
+                Trace.Error(String.Format($"The autologon user '{logonAccount}' doesnt have a user profile on the machine. Please login once with the expected autologon user and reconfigure the agent again"));
```

Comment:
This seems to be an actionable error message, but is buried to the diag log instead of surfaced to the user.